jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -47,6 +47,18 @@ JERRY_STATIC_ASSERT ((int32_t) ECMA_STRING_MAX_CONCATENATION_LENGTH ==
                      ECMA_STRING_MAX_CONCATENATION_LENGTH_should_be_representable_with_int32_t);
 
 /**
+ * The ecma string ref counter should start after the container field.
+ */
+JERRY_STATIC_ASSERT (ECMA_STRING_CONTAINER_MASK + 1 == ECMA_STRING_REF_ONE,
+                     ecma_string_ref_counter_should_start_after_the_container_field);
+
+/**
+ * The ecma string ref and container fields should fill the 16 bit field.
+ */
+JERRY_STATIC_ASSERT ((ECMA_STRING_MAX_REF | ECMA_STRING_CONTAINER_MASK) == UINT16_MAX,
+                     ecma_string_ref_and_container_fields_should_fill_the_16_bit_field);
```

Comment:
A new enum is certainly not needed. But you can add a new label to the end of the existing `ecma_string_container_t` to get the count of labels automatically calculated. I've seen that used in some other enums as well (and called something like `xxx__COUNT`).